hls4ml/utils/config.py

```diff
@@ -321,6 +323,77 @@ def config_from_pytorch_model(
     model_config['Strategy'] = 'Latency'
 
     config['Model'] = model_config
+    config['PytorchModel'] = model
+    config['InputShape'] = input_shape
+
+    if granularity.lower() not in ['model', 'type', 'name']:
+        raise Exception(
+            f'Invalid configuration granularity specified, expected "model", "type" or "name" got "{granularity}"'
+        )
+
+    if backend is not None:
+        backend = hls4ml.backends.get_backend(backend)
+
+    (
+        layer_list,
+        _,
+    ) = hls4ml.converters.parse_pytorch_model(config)
+
+    def make_layer_config(layer):
+        cls_name = layer['class_name']
+        if 'config' in layer.keys():
+            if 'activation' in layer['config'].keys():
+                if layer['config']['activation'] == 'softmax':
+                    cls_name = 'Softmax'
+
+        layer_cls = hls4ml.model.layers.layer_map[cls_name]
+        if backend is not None:
+            layer_cls = backend.create_layer_class(layer_cls)
+
+        layer_config = {}
+
+        config_attrs = [a for a in layer_cls.expected_attributes if a.configurable]
+        for attr in config_attrs:
+            if isinstance(attr, hls4ml.model.attributes.TypeAttribute):
+                precision_cfg = layer_config.setdefault('Precision', {})
+                name = attr.name
+                if name.endswith('_t'):
+                    name = name[:-2]
+                if attr.default is None:
+                    precision_cfg[name] = default_precision
+                else:
+                    precision_cfg[name] = str(attr.default)
+            else:
+                if attr.default is not None:
+                    layer_config[attr.config_name] = attr.default
+
```

Comment:
One bug we recently discovered in the keras parser is that the default reuse factor is not set properly. I didn't try here, but for keras if we set the backend to a real backend, the per-layer reuse factor was always set to 1. See the addition I made in #1016 to fix that. (This PR doesn't change the default to auto, but nevertheless, the reuse factor part is a bug fix, which maybe I should have extracted and not made part of #1016).

Reply:
Thanks for the pointer. Do I understand correctly that the fix is these two lines? https://github.com/fastmachinelearning/hls4ml/pull/1016/files#diff-8ec43234900e263f95104d90f7969c22a9d314817d20ce4d1f23f1a4354fc91aR191-R192

Reply:
Yes.